### DISCUSSION TITLE
More intuitive editing of lists items and their content

### DISCUSSION THREAD

**Discussion by Nikey9** [ScratchAddons/ScratchAddons#1518] [Ideas]
Hello there!
I hope this fits into the ideas category, it's something I have in mind that I think could be extremely helpful (especially for bigger projects), but I'm open to discuss the further details on how exactly it could be implemented for it to make the most sense.

The basic idea is to have a more accessible and intuitive GUI for editing the items in a list and their content.
I'm thinking about a feature where you can right click a list to select "edit", which opens a list editor (in a seperate tab?) of the script editor. 

**The most important features to me would be:** 
- A less clunky feel when editing the text, especially while having multiple lists open (I hope I'm not the only one that feels like this is the case)
- A bigger space to edit list items and the entire text of a list item being displayed (this could be toggled on/off?)
- rearranging the list items through drag and drop

I've been working on a larger game for a while now, with some engines making great use of lists with lots of content per item, so the idea for editing lists inside the scripting area popped into my mind. Now I just hope that someone much smarter than me feels the same hahaha

(ps: Scratch Addons has been insanely helpful to me already and I'm super thankful for it, so many thanks to all the contributors!)

**Comment by pufferfish101007** [comment]
Hmm... maybe a `lists` tab where the code, costumes and sounds are? And then a sidebar with a list of the lists, like the lists of costumes/sounds, and then tools for editing it maybe... this sounds like a really great idea, and might be useful for variables as well, so you don't have to use a `set [var v] to ()` block - maybe that would be another tab?

**Reply by jeffalo** [reply]
perhaps it could be called "variables" then it could also be used to edit variables, cloud variables, and lists. could be pretty useful

**Reply by Nikey9** [reply]
Yeah, a seperate tab would be extremely useful! Just not sure how difficult that is to implement, all things considered.

**Reply by jeffalo** [reply]
#1615
